File: Leetcode/StudyPlanAlgorithm/FloodFill.java
// LeetCode 2021 . 12 . 19
// Study Plan - Algorithm Day 7 733. Flood Fill
package Leetcode.StudyPlanAlgorithm;

import java.util.Arrays;

class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        int m = image[0].length;
        int n = image.length;
        int color = image[sr][sc];
        int [] stack = new int[10];
        
        boolean [][] visited = new boolean[m][n];
        for(boolean a[] :visited) {
            Arrays.fill(a, false);
        }
        
        while(true) {
            int x = 1;
            int y = 2;
            int [][] dir = {{x+1, y}, {x, y+1}, {x-1, y}, {x, y-1}};

            for (int d[] : dir) {
                if(!visited[d[0]][d[1]] && image[d[0]][d[1]] == color) {
                
                }
            }
        }
    }
    
   
}

File: Leetcode/StudyPlanAlgorithm/FloodFill.py
# LeetCode 2021 . 12 . 20
# Study Plan - Algorithm Day 7 733. Flood Fill
from collections import deque

class Solution(object):
    def floodFill(self, image, sr, sc, newColor):
        m = len(image)
        n = len(image[0])
        color = image[sr][sc]
        queue = deque([[sr, sc]])
        visited = [[False] * n for _ in range(m)]
        
        while(len(queue) != 0):
            def direction(loc):
                if(loc[0] >= 1): queue.append([loc[0]-1, loc[1]])
                if(loc[0]+1 < m): queue.append([loc[0]+1, loc[1]])
                if(loc[1] >= 1): queue.append([loc[0], loc[1]-1])
                if(loc[1]+1 < n): queue.append([loc[0], loc[1]+1])

            tmp = queue.popleft()
            if (not visited[tmp[0]][tmp[1]] and image[tmp[0]][tmp[1]] == color):
                image[tmp[0]][tmp[1]] = newColor
                visited[tmp[0]][tmp[1]] = True
                direction(tmp)
 
        return image

sol = Solution()
print(sol.floodFill([[1,1,1],[1,1,0],[1,0,1]], 1, 1, 2))